assets/src/scripts/components/hydrograph/tooltip.js

```diff
@@ -107,63 +173,41 @@ const createTooltip = function(elem, {xScale, yScale, compareXScale, currentTsDa
+        xScale: () => xScale
+    })));
+    focusCurrentCircle.call(link(updateFocusCircle, createStructuredSelector({
+        tsDatum : tsDatumSelector('current'),
+        xScale: () => xScale,
+        yScale: () => yScale
+    })));
+    focusCompareCircle.call(link(updateFocusCircle, createStructuredSelector({
+        tsDatum: tsDatumSelector('compare'),
+        xScale: () => compareXScale,
+        yScale: () => yScale
+    })));
 
     elem.append('rect')
         .attr('class', 'overlay')
         .attr('width', '100%')
         .attr('height', '100%')
-        .on('mouseover', () => {
-            focusLine.style('display', null);
-            focusCurrentCircle.style('display', null);
-            if (isCompareVisible) {
-                focusCompareCircle.style('display', null);
-            }
-            tooltipText.style('display', null);
-        })
-        .on('mouseout', () => {
-            focusLine.style('display', 'none');
-            focusCurrentCircle.style('display', 'none');
-            focusCompareCircle.style('display', 'none');
-            tooltipText.style('display', 'none');
-        })
-        .on('mousemove', function() {
-            const currentTime = xScale.invert(mouse(this)[0]);
-            const currentData = getNearestTime(currentTsData, currentTime);
-
-            if (!currentData) {
-                return;
-            }
-
-            // Update the focus line
-            const currentTimeRange = xScale(currentData.datum.time);
-            focusLine.select('.focus-line')
-                .attr('x1', currentTimeRange)
-                .attr('x2', currentTimeRange);
-
-            // Update the current TS focus circle and tooltip text
-            updateCircleFocus(focusCurrentCircle, {
-                xScale: xScale,
-                yScale: yScale,
-                tsDatum: currentData.datum
-            });
-            updateTooltipText(tooltipText.select('.current-tooltip-text'), currentData.datum);
-
-            //Update the compare TS focus circle and tooltip text if we are showing it.
-            if (isCompareVisible) {
-                const compareTime = compareXScale.invert(mouse(this)[0]);
-                const compareData = getNearestTime(compareTsData, compareTime);
-                if(!compareData) {
-                    return;
-                }
-                updateTooltipText(tooltipText.select('.compare-tooltip-text'), compareData.datum);
-                updateCircleFocus(focusCompareCircle, {
-                    xScale: compareXScale,
-                    yScale: yScale,
-                    tsDatum: compareData.datum
-                });
-            }
-        });
+        .on('mouseover', dispatch(function() {
+            return Actions.setTooltipTime(
+                xScale.invert(mouse(elem.node())[0]),
+                isCompareVisible ? compareXScale.invert(mouse(elem.node())[0]) : null
+            );
```

Comment:
It's too bad the scales need to be different for each line, but given that I'm thinking doing the xScale.invert() in the code that moves the tooltip circle would be a (tiny) bit simpler, so only the single mouse location needs to be stored in state.

Reply:
It's true that I could put the mouse location, but given that the tooltip text is in a different group then the tooltip focus, I think it's cleare to store the actual time.